src/main/java/core/basesyntax/Car.java

```diff
@@ -1,13 +1,89 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
-/**
- * Make this class immutable. See requirements in the README file
- */
-public class Car {
-    private int year;
-    private String color;
-    private List<Wheel> wheels;
-    private Engine engine;
+public final class Car {
+    private final int year;
+    private final String color;
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        this.year = year;
+        this.color = color;
+        this.wheels = cloneWheels(wheels);
+        this.engine = engine == null ? null : engine.clone();
+    }
+
+    public int getYear() {
+        return year;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public List<Wheel> getWheels() {
+        return cloneWheels(wheels);
+    }
+
+    public Engine getEngine() {
+        return engine == null ? null : engine.clone();
+    }
+
+    public Car changeEngine(Engine newEngine) {
+        return new Car(year, color, wheels, newEngine);
+    }
+
+    public Car changeColor(String newColor) {
+        return new Car(year, newColor, wheels, engine);
+    }
+
+    public Car addWheel(Wheel newWheel) {
+        List<Wheel> clonedListOfWheels = cloneWheels(wheels);
```

Comment:
```suggestion
        List<Wheel> clonedListOfWheels = new ArrayList(wheels);
```
you gonna clone them in the constructor anyway